src/core/manager.c

```diff
@@ -2129,6 +2129,18 @@ void manager_clear_jobs(Manager *m) {
                 job_finish_and_invalidate(j, JOB_CANCELED, false, false);
 }
 
+void manager_unwatch_pid(Manager *m, pid_t pid) {
+        Unit **array;
+
+        /* First let's drop the unit keyed as "pid". */
+        (void) hashmap_remove(m->watch_pids, PID_TO_PTR(pid));
+
+        /* Then, let's also drop the array keyed by -pid. */
+        array = hashmap_remove(m->watch_pids, PID_TO_PTR(-pid));
+        if (array)
+                free(array);
```

Comment:
free() is actually happy with NULL ptrs and is a NOP then. Hence please shorten to:

```c
free(hashmap_remove(…));
```